common/types/encode.go

```diff
@@ -62,41 +61,16 @@ func NIPSTChallengeAsBytes(challenge *NIPSTChallenge) ([]byte, error) {
 	return w.Bytes(), nil
 }
 
-func AddressableTransactionAsBytes(tx *AddressableSignedTransaction) ([]byte, error) {
-	var w bytes.Buffer
-	if _, err := xdr.Marshal(&w, tx); err != nil {
-		return nil, fmt.Errorf("error marshalling transaction: %v", err)
-	}
-	return w.Bytes(), nil
-}
-
-func BytesAsAddressableTransaction(buf []byte) (*AddressableSignedTransaction, error) {
-	b := AddressableSignedTransaction{}
+func BytesAsTransaction(buf []byte) (*Transaction, error) {
```

Comment:
This is sugar, it returns a typed variable instead of the caller needing to create a variable and pass a ref into `BytesToInterface` for it to be filled. It has an additional advantage that developers can easily find all occurrences of tx deserialization, instead of having to sift through **all** deserializations (of any type).